libsyclinterface/source/dpctl_sycl_extension_interface.cpp

```diff
@@ -62,3 +62,37 @@ bool DPCTLWorkGroupMemory_Available()
     return false;
 #endif
 }
+
+DPCTL_API
+__dpctl_give DPCTLSyclRawKernelArgRef DPCTLRawKernelArg_Create(void *bytes,
+                                                               size_t count)
+{
+    DPCTLSyclRawKernelArgRef rka = nullptr;
+    try {
+        auto rawData =
+            std::unique_ptr<unsigned char[]>(new unsigned char[count]);
+        std::memcpy(rawData.get(), bytes, count);
+        auto RawKernelArg = std::unique_ptr<RawKernelArgData>(
+            new RawKernelArgData{rawData.release(), count});
+        rka = wrap<RawKernelArgData>(RawKernelArg.release());
+    } catch (std::exception const &e) {
+        error_handler(e, __FILE__, __func__, __LINE__);
+    }
+    return rka;
+}
+
+DPCTL_API
+void DPCTLRawKernelArg_Delete(__dpctl_take DPCTLSyclRawKernelArgRef Ref)
+{
+    delete unwrap<RawKernelArgData>(Ref);
```

Comment:
```C++
auto RawArg = unwrap<RawKernelArgData>(Ref);
delete[] RawArg->bytes;
delete RawArg;
```

Reply:
I changed the approach slightly to hold the data in a `std::vector` inside `RawKernelArgDataTy`, because `delete[] RawArg->bytes` gave a warning about incomplete delete.